src/mixins/java/org/spongepowered/common/mixin/core/world/entity/EntityMixin.java

```diff
@@ -265,52 +293,42 @@
-                        repositionEvent.destinationPosition().y(), repositionEvent.destinationPosition().z());
+                final net.minecraft.server.level.ServerLevel originalWorld = (net.minecraft.server.level.ServerLevel) this.shadow$getCommandSenderWorld();
+                ((PlatformServerLevelBridge) this.shadow$getCommandSenderWorld()).bridge$removeEntity((Entity) (Object) this, true);
+                this.bridge$revive();
+                this.shadow$setLevel(destinationWorld);
+                destinationWorld.addFromAnotherDimension((Entity) (Object) this);
+
+                originalWorld.resetEmptyTime();
+                destinationWorld.resetEmptyTime();
             } else {
-                final Vector3d destination;
                 if (ShouldFire.MOVE_ENTITY_EVENT) {
                     final MoveEntityEvent event = SpongeEventFactory.createMoveEntityEvent(frame.currentCause(),
                             (org.spongepowered.api.entity.Entity) this, VecHelper.toVector3d(this.shadow$position()),
                             location.position(), location.position());
                     if (SpongeCommon.postEvent(event)) {
                         return false;
                     }
-                    destination = event.destinationPosition();
+                    destinationPosition = event.destinationPosition();
                 } else {
-                    destination = location.position();
+                    destinationPosition = location.position();
                 }
-                this.shadow$setPos(destination.x(), destination.y(), destination.z());
-            }
 
-            if (!destinationWorld.getChunkSource().hasChunk((int) this.shadow$getX() >> 4, (int) this.shadow$getZ() >> 4)) {
-                // Roll back the position
-                this.shadow$setPos(originalPosition.x, originalPosition.y, originalPosition.z);
-                return false;
+                ((Entity) (Object) this).unRide();
             }
-
-            ((Entity) (Object) this).unRide();
-
-            final net.minecraft.server.level.ServerLevel originalWorld = (net.minecraft.server.level.ServerLevel) this.shadow$getCommandSenderWorld();
-            ((PlatformServerLevelBridge) this.shadow$getCommandSenderWorld()).bridge$removeEntity((Entity) (Object) this, true);
-            this.bridge$revive();
-            this.shadow$setLevel(destinationWorld);
-            destinationWorld.addFromAnotherDimension((Entity) (Object) this);
-
-            originalWorld.resetEmptyTime();
-            destinationWorld.resetEmptyTime();
-
-            final ChunkPos chunkPos = new ChunkPos((int) this.shadow$getX() >> 4, (int) this.shadow$getZ() >> 4);
-            destinationWorld.getChunkSource().addRegionTicket(TicketType.POST_TELEPORT, chunkPos, 1, ((Entity) (Object) this).getId());
+            this.shadow$teleportToWithTicket(destinationPosition.x(), destinationPosition.y(), destinationPosition.z());
```

Comment:
Same problem again here - no need to fire the event above if it's going to happen in this method.